commit 3faa51a0caebb56746988ae7f5e9e9f649dc42b6
Author: nicm <nicm>
Date:   Sun Oct 25 08:59:26 2015 +0000

    Pass output from jobs through format_expand() so they are expanded again
    (this was the previous behaviour).

--- a/format.c
+++ b/format.c
@@ -40,7 +40,7 @@
 typedef void (*format_cb)(struct format_tree *, struct format_entry *);
 
 void	 format_job_callback(struct job *);
-const char *format_job_get(struct format_tree *, const char *);
+char	*format_job_get(struct format_tree *, const char *);
 void	 format_job_timer(int, short, void *);
 
 void	 format_cb_host(struct format_tree *, struct format_entry *);
@@ -214,7 +214,7 @@
 }
 
 /* Find a job. */
-const char *
+char *
 format_job_get(struct format_tree *ft, const char *cmd)
 {
 	struct format_job	fj0, *fj;
@@ -244,7 +244,7 @@
 	if (ft->flags & FORMAT_STATUS)
 		fj->status = 1;
 
-	return (fj->out);
+	return (format_expand(ft, fj->out));
 }
 
 /* Remove old jobs. */
@@ -726,9 +726,9 @@
 char *
 format_expand(struct format_tree *ft, const char *fmt)
 {
-	char		*buf, *tmp, *cmd;
+	char		*buf, *tmp, *cmd, *out;
 	const char	*ptr, *s, *saved = fmt;
-	size_t		 off, len, n, slen;
+	size_t		 off, len, n, outlen;
 	int     	 ch, brackets;
 
 	if (fmt == NULL)
@@ -768,18 +768,20 @@
 			tmp[n] = '\0';
 			cmd = format_expand(ft, tmp);
 
-			s = format_job_get(ft, cmd);
-			slen = strlen(s);
+			out = format_job_get(ft, cmd);
+			outlen = strlen(out);
 
 			free(cmd);
 			free(tmp);
 
-			while (len - off < slen + 1) {
+			while (len - off < outlen + 1) {
 				buf = xreallocarray(buf, 2, len);
 				len *= 2;
 			}
-			memcpy(buf + off, s, slen);
-			off += slen;
+			memcpy(buf + off, out, outlen);
+			off += outlen;
+
+			free(out);
 
 			fmt += n + 1;
 			continue;
